polyply/tests/test_build_file_parser.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test that build files are properly read.
+"""
+import textwrap
+import pytest
+import numpy as np
+import vermouth.forcefield
+import vermouth.molecule
+import polyply
+from polyply.src.meta_molecule import Monomer, MetaMolecule
+from polyply.src.topology import Topology
+
+@pytest.mark.parametrize('tokens, expected', (
+   # for cylinder
+   (["PEO", "63", "154", "in", "5", "5", "5", "5", "5"],
+    {"resname": "PEO", "start": 63, "stop": 154, "parameters":["in", np.array([5.0, 5.0, 5.0]), 5.0, 5.0]})
+   # for recangle
+   ,(["PEO", "0", "10", "out", "10", "10", "10", "5", "5", "3"],
+    {"resname": "PEO", "start": 0, "stop": 10, "parameters":["out", np.array([10.0, 10.0, 10.0]), 5.0, 5.0, 3.0]})
+   # for sphere
+   ,(["PEO", "0", "10", "in", "10", "10", "10", "5"],
+    {"resname": "PEO", "start": 0, "stop": 10, "parameters":["in", np.array([10.0, 10.0, 10.0]), 5.0]})
+   ))
+def test_base_parser_geometry(tokens, expected):
+    result = polyply.src.build_file_parser.BuildDirector._base_parser_geometry(tokens)
+    assert result.keys() == expected.keys()
+    for key in result:
+        if key != "parameters":
+            assert result[key] == expected[key]
+        else:
+            assert result[key][0] == expected[key][0]
+            assert all(result[key][1] == expected[key][1])
+            for result_param, expected_param in zip(result[key][2:], expected[key][2:]):
+                assert result_param == expected_param
+
+@pytest.fixture
+def test_molecule():
```

Comment:
As before.
Consider importing these fixtures from a helper file if they're reusable.

Reply:
I'll note this down for a cleaning PR on the tests. At the moment this isn't essential and the various molecules and test systems have slightly different complexity depending on what the test demands.